tools/Custom/HttpMessageFormatter.cs

```diff
@@ -0,0 +1,356 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph.PowerShell
+{
+    /// <summary>
+    /// Derived <see cref="HttpContent"/> class which can encapsulate an <see cref="HttpResponseMessage"/>
+    /// or an <see cref="HttpRequestMessage"/> as an entity with media type "application/http".
+    /// </summary>
+    internal class HttpMessageFormatter : HttpContent
+    {
+        private const string SP = " ";
+        private const string ColonSP = ": ";
+        private const string CRLF = "\r\n";
+        private const string CommaSeparator = ", ";
+
+        private const int DefaultHeaderAllocation = 2 * 1024;
+
+        private const string DefaultMediaType = "application/http";
+
+        private const string MsgTypeParameter = "msgtype";
+        private const string DefaultRequestMsgType = "request";
+        private const string DefaultResponseMsgType = "response";
+
+        // Set of header fields that only support single values such as Set-Cookie.
+        private static readonly HashSet<string> _singleValueHeaderFields = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "Cookie",
+            "Set-Cookie",
+            "X-Powered-By",
+        };
+
+        // Set of header fields that should get serialized as space-separated values such as User-Agent.
+        private static readonly HashSet<string> _spaceSeparatedValueHeaderFields = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "User-Agent",
+        };
+
+        // Set of header fields that should not get serialized
+        private static readonly HashSet<string> _neverSerializedHeaderFields = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "SdkVersion",
```

Comment:
Why are we not Serializing SdkVersion and FeatureFlag? Or Cache-Control, or Date?

Reply:
This is legacy from the contrib project.

Reply:
We should serialize headers that are useful in tracing an issue/bug. For instance, we need SdkVersion to know which version of the SDK the customer was running, featureFlag to know which middleware handlers they have enabled e.t.c. `Authorization` header is not useful, and we shouldn't serialize it.